src/clic.adoc

```diff
@@ -505,34 +505,36 @@ The conditions for an interrupt trap to occur must be evaluated in a bounded amo
 from when an interrupt becomes, or ceases to be, pending in `clicintip`, but unlike the {ip}/{ie} CSRs, there is no requirement
 that `clicintie` or `clicintip` are evaluated immediately following an explicit store to `clicintip` or `clicintie`.
 
-When the input is configured for level-sensitive input, the
-`clicintip[__i__]` bit reflects the value of an input signal to the
+When the input is configured for edge-sensitive input,
+`clicintip[__i__]` is a read-write bit that can be updated both
+by hardware interrupt inputs and by software. The bit is set by
+hardware after an edge of the appropriate polarity is observed on the
+interrupt input, as determined by the `clicintattr[__i__]` field.
+
+When the input is configured for level-sensitive input,
+each `clicintip[__i__]` may be writable or read-only.
+When read-only, `clicintip[__i__]` bit reflects the value of an input signal to the
 interrupt controller after any conditional inversion specified by the
 `clicintattr[__i__]` field, and software writes to the bit are ignored.
 Software clears the interrupt at the source device.
+When writable, additionally, the software-writable is applied with a logical-OR.
+Only the software-writable bit participates in
+the read-modify-write sequence of a CSRRS or CSRRC instruction.
 
-When the input is configured for edge-sensitive input,
-`clicintip[__i__]` is a read-write register that can be updated both
-by hardware interrupt inputs and by software.  The bit is set by
-hardware after an edge of the appropriate polarity is observed on the
-interrupt input, as determined by the `clicintattr[__i__]` field.
-Software writes to `clicintip[__i__]` can set or
-clear edge-triggered pending bits directly by writes to the
+Note: As per the machine-level ISA, this behavior is required for the SEI interrupt (exccode 9).
 
-`clicintip[__i__]` register. Edge-triggered pending bits can also be cleared when a CSR instruction that accesses {nxti} includes a write.
-`clicintip[__i__]` behavior is unaffected by `clicintie[__i__]` setting.
+If `clicintip[__i__]` is writable, a pending interrupt i can be cleared by writing 0 to this bit.
```

Comment:
If the input wire is still 1 it wouldn't be cleared though right? Since the software bit is ORed with it?

Reply:
Agreed with @Timmmm. Shouldn't this state that interrupt clearing might also need to be performed at the very source (driving the interrupt wire)?